File: WhosWill/WhosWill/Models/SkillsData.swift
//
//  SkillsData.swift
//  WhosWill
//
//  Static skill categories and languages (labels come from translations).
//

import Foundation

struct SkillItem: Identifiable {
    let id = UUID()
    let name: String
    let level: Int // 0–100
}

struct SkillCategory: Identifiable {
    let id = UUID()
    let key: String // "mobile" | "ai" | "web" | "tools"
    let items: [SkillItem]
}

struct LanguageItem: Identifiable {
    let id = UUID()
    let nameKey: String
    let proficiencyKey: String
    let level: Int
}

enum SkillsData {
    static let categories: [SkillCategory] = [
        SkillCategory(key: "mobile", items: [
            SkillItem(name: "React Native", level: 92),
            SkillItem(name: "Swift / iOS", level: 88),
            SkillItem(name: "Kotlin / Android", level: 85),
            SkillItem(name: "Flutter", level: 75),
        ]),
        SkillCategory(key: "ai", items: [
            SkillItem(name: "Computer vision (on-device)", level: 88),
            SkillItem(name: "Machine learning", level: 75),
            SkillItem(name: "Python", level: 82),
        ]),
        SkillCategory(key: "web", items: [
            SkillItem(name: "React / React.js", level: 85),
            SkillItem(name: "Node.js", level: 80),
            SkillItem(name: "Laravel", level: 70),
        ]),
        SkillCategory(key: "tools", items: [
            SkillItem(name: "Git & version control", level: 90),
            SkillItem(name: "Jira & Scrum", level: 85),
            SkillItem(name: "DevOps / deployment", level: 78),
            SkillItem(name: "MVVM / OOP", level: 88),
        ]),
    ]
    
    static let languages: [LanguageItem] = [
        LanguageItem(nameKey: "english", proficiencyKey: "fullProfessional", level: 90),
        LanguageItem(nameKey: "cantonese", proficiencyKey: "native", level: 100),
        LanguageItem(nameKey: "mandarin", proficiencyKey: "fullProfessional", level: 88),
    ]
    
    static let yearsExperience = 5
}

File: WhosWillAndroid/app/src/main/java/com/example/willlam/whoswill/data/SkillsData.kt
package com.example.willlam.whoswill.data

data class SkillItem(val name: String, val level: Int)

data class SkillCategory(val key: String, val items: List<SkillItem>)

data class LanguageItem(val nameKey: String, val proficiencyKey: String, val level: Int)

object SkillsData {
    val categories: List<SkillCategory> = listOf(
        SkillCategory("mobile", listOf(
            SkillItem("React Native", 92),
            SkillItem("Swift / iOS", 88),
            SkillItem("Kotlin / Android", 85),
            SkillItem("Flutter", 75)
        )),
        SkillCategory("ai", listOf(
            SkillItem("Computer vision (on-device)", 88),
            SkillItem("Machine learning", 75),
            SkillItem("Python", 82)
        )),
        SkillCategory("web", listOf(
            SkillItem("React / React.js", 85),
            SkillItem("Node.js", 80),
            SkillItem("Laravel", 70)
        )),
        SkillCategory("tools", listOf(
            SkillItem("Git & version control", 90),
            SkillItem("Jira & Scrum", 85),
            SkillItem("DevOps / deployment", 78),
            SkillItem("MVVM / OOP", 88)
        ))
    )

    val languages: List<LanguageItem> = listOf(
        LanguageItem("english", "fullProfessional", 90),
        LanguageItem("cantonese", "native", 100),
        LanguageItem("mandarin", "fullProfessional", 88)
    )

    const val yearsExperience = 5
}
